src/main/java/com/onarandombox/MultiverseCore/commands/ListCommand.java

```diff
@@ -43,52 +41,52 @@ public ListCommand(MultiverseCore plugin) {
 
     @Override
     public void runCommand(CommandSender sender, List<String> args) {
-        ContentFilter filter = ContentFilter.DEFAULT;
+        ContentFilter filter = DefaultContentFilter.INSTANCE;
         int page = 1;
 
         // Either page or filter.
         if (args.size() == 1) {
             try {
                 page = Integer.parseInt(args.get(0));
             } catch (NumberFormatException ignore) {
-                filter = new ContentFilter(args.get(0));
+                filter = RegexContentFilter.fromString(args.get(0));
             }
         }
 
         // Filter then page.
         if (args.size() == 2) {
-            filter = new ContentFilter(args.get(0));
+            filter = RegexContentFilter.fromString(args.get(0));
             try {
                 page = Integer.parseInt(args.get(1));
             } catch (NumberFormatException ignore) {
                 sender.sendMessage(ChatColor.RED + args.get(1) + " is not valid number!");
             }
         }
 
-        ContentDisplay.forContent(getListContents(sender))
-                .header("%s====[ Multiverse World List ]====", ChatColor.GOLD)
-                .displayHandler(DisplayHandlers.PAGE_LIST)
-                .colorTool(ColorAlternator.with(ChatColor.AQUA, ChatColor.GOLD))
-                .filter(filter)
-                .setting(PagedDisplaySettings.SHOW_PAGE, page)
-                .show(sender);
+        ContentDisplay.create()
+                .addContentParser(newWorldListContentParser())
+                .withSendHandler(PagedSendHandler.create()
+                        .withHeader("%s====[ Multiverse World List ]====", ChatColor.GOLD)
+                        .withFilter(filter)
+                        .withTargetPage(page))
+                .send(sender);
     }
 
-    private Collection<String> getListContents(@NotNull CommandSender sender) {
-        Player player = (sender instanceof Player) ? (Player) sender : null;
+    private ContentParser newWorldListContentParser() {
```

Comment:
I think that this might be more readable as a separate class. It could just be a private static one in this class maybe?

Reply:
yea can change that. Probably use a singleton instance?

Reply:
hmm dont really want do use static class so I don't have to pass in the `plugin` instance to the private class, so current just doing something like `private class WorldListContentParser implements ContentParser {` and then `.addContentParser(new WorldListContentParser());`

Reply:
What's wrong with passing the plugin instance into the new class's constructor?

Reply:
tbh nothing wrong, but then I can't really use a single instance anyways, as not a good idea to store plugin objects in static like this right? Like if a new instance plugin instance is created, it may cause issues?